### PR TITLE
Add test for migrating DeclStmt inside lambda expression

### PR DESCRIPTION
**Pull request by delecui**

### CODE REVIEW

**Comment by delecui** [comment]
This test patch is for bug fix in https://github.com/oneapi-src/SYCLomatic/pull/55

File: features/test_feature.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
              'thrust-qmc', 'thrust-transform-if', 'thrust-policy', 'thrust-list', 'module-kernel',
              'kernel-launch', 'thrust-gather', 'thrust-scatter', 'thrust-unique_by_key_copy', 'thrust-for-hypre',
              'thrust-rawptr-noneusm', 'driverStreamAndEvent', 'grid_sync', 'deviceProp', 'cub_block_p2',
-             'cub_device', 'activemask', 'complex']
+             'cub_device', 'activemask', 'complex', 'user_defined_rules']
```
Comment (delecui) [inline features/test_feature.py:23]: mark "user_defined_rules" as execution test to enbale linkage check.
the replaced API "my_min" and "foo3" are changed to "extern" functions. 
So link error will happen if rules "foo3" and "min" do not work.

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by ShengchenJ — APPROVED** [review]
LGTM

**Pull request merged**